### PR TITLE
Add upload mechanism to stub branch

### PR DESCRIPTION
**Pull request by ZachCMP**
https://www.pivotaltracker.com/story/show/183190260

### CODE REVIEW

**Review by eanders — APPROVED** [review]
This is great.  
I believe it should be merged after
https://github.com/greenriver/hmis-warehouse/pull/2683
and 
https://github.com/greenriver/hmis-warehouse/pull/2680

File: drivers/hmis/app/models/hmis/hud/client.rb [hunk 1]
```diff
@@ -28,6 +28,16 @@ class Hmis::Hud::Client < Hmis::Hud::Base
 
   validates_with Hmis::Hud::Validators::ClientValidator
 
+  # ! Elliot: This logic probably wouldn't live here, but I've included it here for reference
```
Comment (eanders) [inline drivers/hmis/app/models/hmis/hud/client.rb:33]: @eanders needs to setup
Client has_one client_headshot (ClientFile) with appropriate tag.

@ZachCMP can you add generic file upload backed by ClientFile, these will also need to include "tags" and note

File: drivers/hmis/app/graphql/types/hmis_schema/client_input.rb [hunk 2]
```diff
@@ -20,5 +20,6 @@ class HmisSchema::ClientInput < BaseInputObject
     argument :race, [Types::HmisSchema::Enums::Race], required: false
     argument :ethnicity, Types::HmisSchema::Enums::Hud::Ethnicity, required: false
     argument :veteran_status, Types::HmisSchema::Enums::Hud::NoYesReasonsForMissingData, required: false
+    argument :image_blob_id, ID, required: false
```
Comment (ZachCMP) [inline drivers/hmis/app/graphql/types/hmis_schema/client_input.rb:22]: - [ ] Add `ClientFileInputType` with tags, blob_id and note

File: drivers/hmis/app/models/hmis/hud/client.rb (hunk 1 above)
Comment (eanders) [reply drivers/hmis/app/models/hmis/hud/client.rb:33]: This should look something like `set_local_client_image_cache`

```
file = GrdaWarehouse::ClientFile.create(
          client_id: id,
          user_id: user.id,
          name: 'Client Headshot',
          visible_in_window: false,
        )
file.tag_list.add('Client Headshot')
        file.client_file.attach(io: StringIO.open(image_data), filename: "client_headshot_#{id}")
```

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -153,16 +155,17 @@ services:
...
-      - MINIO_ROOT_PASSWORD=local_secret_key
+      MINIO_ACCESS_KEY: local_access_key
+      MINIO_SECRET_KEY: local_secret_key
+      MINIO_ROOT_USER: local_access_key
+      MINIO_ROOT_PASSWORD: local_secret_key
     ports:
       - 9000:9000
       - 9001:9001 # enable for UI access
-    command: server /data
+    command: server --address :9000 --console-address :9001 --certs-dir /certs /data
```
Comment (eanders) [inline docker/docker-compose.yml]: Note to Elliot, might need to add the ports on this line.  testing.

File: drivers/hmis/app/graphql/types/hmis_schema/client_input.rb (hunk 2 above)
Comment (ZachCMP) [reply drivers/hmis/app/graphql/types/hmis_schema/client_input.rb:22]: ^^^ I did not do this as part of this work. Does this still need to be done?

File: drivers/hmis/lib/form_data/records/client.json [hunk 4]
```diff
@@ -181,6 +191,29 @@
...
+      ],
+      "item": [
+        {
+          "type": "IMAGE",
+          "required": false,
+          "link_id": "image_blob_id",
+          "text": "Image",
+          "field_name": "imageBlobId"
+        }
+      ]
```
Comment (eanders) [inline drivers/hmis/lib/form_data/records/client.json:216]: I think this should come out.  There's a section on this form that is empty and shouldn't be there.

**Review by eanders — APPROVED** [review]
@ZachCMP The back-end looks good!  The front end needs two things.
1. Delete button to remove an image if accidentally uploaded.
2. A button to open a standard upload field.  As written there's no way to add a photo except drag and drop.

File: drivers/hmis/lib/form_data/records/client.json (hunk 4 above)
Comment (gigxz) [reply drivers/hmis/lib/form_data/records/client.json:216]: That's not what I'm seeing. It looks good to me -- the section appears on the New client form, but not on the Update client form.

There was an issue with the `Image` header incorrectly appearing on the left nav, but that should be fixed with https://github.com/greenriver/hmis-frontend/pull/117

Comment (eanders) [reply drivers/hmis/lib/form_data/records/client.json:216]: Ok, I defer to you!  I just was noting that "Image" link showing up on the client edit screen with no section visible.

**Comment by ZachCMP** [comment]
@eanders I've addressed those things in the frontend PR. However, there's an open question about how we would actually clear an image if it has already been saved to a client (see [this comment](https://github.com/greenriver/hmis-frontend/pull/115#issuecomment-1371484673))

**Comment by eanders** [comment]
@ZachCMP these changes seem to have broken some tests, can you take a look?

**Comment by ZachCMP** [comment]
@eanders I implemented the photo deletion logic discussed in the [frontend PR](https://github.com/greenriver/hmis-frontend/pull/115). Part of the work is in this PR and part of the work is in that PR

**Comment by gigxz** [comment]
@eanders can we merge this in to 61?

**Comment by eanders** [comment]
@gigxz merged

**Pull request merged**